In [12]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [13]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')

In [14]:
df.head()

age  black  case  clerical  construc  educ  earns74  gdhlth  inlf  leis1  \
0   32      0     1       0.0       0.0    12        0       0     1   3529   
1   31      0     2       0.0       0.0    14     9500       1     1   2140   
2   44      0     3       0.0       0.0    17    42500       1     1   4595   
3   30      0     4       0.0       0.0    12    42500       1     1   3211   
4   64      0     5       0.0       0.0    14     2500       1     1   4052   

   ...  spwrk75  totwrk  union  worknrm  workscnd  exper  yngkid  yrsmarr  \
0  ...        0    3438      0     3438         0     14       0       13   
1  ...        0    5020      0     5020         0     11       0        0   
2  ...        1    2815      0     2815         0     21       0        0   
3  ...        1    3786      0     3786         0     12       0       12   
4  ...        1    2580      0     2580         0     44       0       33   

      hrwage  agesq  
0   7.070004   1024  
1   1.429999    961  
2  20.529997   1936  
3   9.619998    900  
4   2.750000   4096  

[5 rows x 34 columns]

In [15]:
# специфицируем модель через формулу
sleep_eq1 = smf.ols(formula='sleep~totwrk+male', data=df).fit()
# Коэфициенты модели с округление
sleep_eq1.params.round(2)

Intercept    3573.20
totwrk         -0.17
male           88.84
dtype: float64

## Спецификация
$$
sleep = 3573.20 - 0.17*totwrk + 88.84*male
$$

## Интерпретация  

1. При увеличении рабочего времени в неделю количество сна уменьшается на 0.17 минуты
2. Мужчины спят на 88.84 минуты в неделю больше женщин

In [16]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [17]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')
df.head()

age  black  case  clerical  construc  educ  earns74  gdhlth  inlf  leis1  \
0   32      0     1       0.0       0.0    12        0       0     1   3529   
1   31      0     2       0.0       0.0    14     9500       1     1   2140   
2   44      0     3       0.0       0.0    17    42500       1     1   4595   
3   30      0     4       0.0       0.0    12    42500       1     1   3211   
4   64      0     5       0.0       0.0    14     2500       1     1   4052   

   ...  spwrk75  totwrk  union  worknrm  workscnd  exper  yngkid  yrsmarr  \
0  ...        0    3438      0     3438         0     14       0       13   
1  ...        0    5020      0     5020         0     11       0        0   
2  ...        1    2815      0     2815         0     21       0        0   
3  ...        1    3786      0     3786         0     12       0       12   
4  ...        1    2580      0     2580         0     44       0       33   

      hrwage  agesq  
0   7.070004   1024  
1   1.429999    961  
2  20.529997   1936  
3   9.619998    900  
4   2.750000   4096  

[5 rows x 34 columns]

In [18]:
# специфицируем модель через формулу
sleep_eq1 = smf.ols(formula='sleep~totwrk+male+smsa+age+south+yngkid+marr+union', data=df).fit()
# Коэфициенты модели с округление
sleep_eq1.params.round(2)

Intercept    3446.83
totwrk         -0.17
male           87.11
smsa          -54.19
age             2.71
south         102.27
yngkid        -13.05
marr           31.36
union          11.87
dtype: float64

## Спецификация 
$$
sleep = 3446.83 - 0.17 * totwrk + 87.11 * male - 54.19 * smsa + 2.71 * age + 102.27 * south - 13.05 * yngkid + 31.36 * marr + 11.87 * union
$$

## Интерпретация

1. При увеличении рабочего времени в неделю количество сна уменьшается на 0.17 минуты
2. Мужчины спят на 88.84 минуты в неделю больше женщин

In [19]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [20]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/sleep75.csv')
df.head()

age  black  case  clerical  construc  educ  earns74  gdhlth  inlf  leis1  \
0   32      0     1       0.0       0.0    12        0       0     1   3529   
1   31      0     2       0.0       0.0    14     9500       1     1   2140   
2   44      0     3       0.0       0.0    17    42500       1     1   4595   
3   30      0     4       0.0       0.0    12    42500       1     1   3211   
4   64      0     5       0.0       0.0    14     2500       1     1   4052   

   ...  spwrk75  totwrk  union  worknrm  workscnd  exper  yngkid  yrsmarr  \
0  ...        0    3438      0     3438         0     14       0       13   
1  ...        0    5020      0     5020         0     11       0        0   
2  ...        1    2815      0     2815         0     21       0        0   
3  ...        1    3786      0     3786         0     12       0       12   
4  ...        1    2580      0     2580         0     44       0       33   

      hrwage  agesq  
0   7.070004   1024  
1   1.429999    961  
2  20.529997   1936  
3   9.619998    900  
4   2.750000   4096  

[5 rows x 34 columns]

In [28]:
# специфицируем модель через формулу
sleep_eq1 = smf.ols(formula='sleep~np.log(hrwage)+smsa+totwrk+male+marr+age+south+yngkid', data=df).fit()
# Коэфициенты модели с округление
sleep_eq1.params.round(2)

Intercept         3440.19
np.log(hrwage)      -1.39
smsa               -36.96
totwrk              -0.16
male                36.87
marr                53.34
age                  2.37
south               76.27
yngkid              47.92
dtype: float64

## Спецификация 

$$
sleep = 3440.19 - 1.39*log(hrwage) - 36.96*smsa - 0.16*totwrk + 36.87*male + 53.34*marr + 2.37*age + 76.27*south + 47.92*yngkid
$$

## Интерпретация 

1. При увеличении почасовой оплаты на 1% количество сна уменьшается на 0.0139 минут
2. Жители мегаполиса спят на 36.96 минут в неделю меньше 